slack/nominate_test.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	if !strings.Contains(sentMessage, "Nobody to nominate") {
+		t.Error("Expected sent message to inform that nobody can be nominated")
+	}
+}
+
+func TestHandleNominateFetchesNominatedUserInformationsAndFailsOnError(t *testing.T) {
+	var (
+		userInConversation = []string{"a", "b", "c"}
+
+		genMaxValue int
+		askedUserID string
+
+		botID   = "johnny"
+		channel = "foochan"
+		caller  = "caller"
+	)
+
+	userInConversation = append(userInConversation, botID)
+
+	fakeClient := &fakeRtmClient{
+		GetUsersInConversationHandler: func(q *slack.GetUsersInConversationParameters) ([]string, string, error) {
+			return userInConversation, "", nil
+		},
+		GetInfoHandler: func() *slack.Info {
+			return &slack.Info{User: &slack.UserDetails{ID: botID}}
+		},
+		GetUserInfoHandler: func(userID string) (*slack.User, error) {
+			askedUserID = userID
+			return nil, errors.New("Nope")
+		},
+	}
+
+	fakeGenerator := &fakeIntGenerator{
+		IntnHandler: func(maxVal int) int {
+			genMaxValue = maxVal
+			return 0
+		},
+	}
+
+	bot := rtmBot{
+		rtm:    fakeClient,
+		gen:    fakeGenerator,
+		logger: log.New(ioutil.Discard, "testmusicof-bot: ", log.Lshortfile|log.LstdFlags),
+	}
+
+	err := bot.handleNominate(caller, channel)
+
+	if err == nil {
+		t.Fatal("Expected an error, got nothing")
+	}
+
+	if genMaxValue != len(userInConversation)-1 {
+		t.Error("Supposed to ask to generator a max value equals to the length of the user in conversation got", genMaxValue)
+	}
+
+	if askedUserID != "a" {
+		t.Error("Wrong userID asked, expected a got", askedUserID)
+	}
+}
+
+func TestHandleNominateNotifiesNominatedUser(t *testing.T) {
+	var (
+		userInConversation = []string{"a", "b", "c"}
+
+		selectedUserName = "michel"
+		sentChannel      string
+		sentMessage      string
+
+		botID   = "johnny"
+		channel = "foochan"
+		caller  = "caller"
+	)
+
+	userInConversation = append(userInConversation, botID)
+
+	fakeClient := &fakeRtmClient{
+		GetUsersInConversationHandler: func(q *slack.GetUsersInConversationParameters) ([]string, string, error) {
+			return userInConversation, "", nil
+		},
+		GetInfoHandler: func() *slack.Info {
+			return &slack.Info{User: &slack.UserDetails{ID: botID}}
+		},
+		GetUserInfoHandler: func(userID string) (*slack.User, error) {
+			return &slack.User{Name: selectedUserName}, nil
+		},
+		PostMessageHandler: func(channel string, message string, params slack.PostMessageParameters) (string, string, error) {
+			sentChannel = channel
+			sentMessage = message
+			return "", "", nil
+		},
+	}
+
+	fakeGenerator := &fakeIntGenerator{
+		IntnHandler: func(maxVal int) int {
+			return 0
+		},
+	}
+
+	bot := rtmBot{
+		rtm:    fakeClient,
+		gen:    fakeGenerator,
+		logger: log.New(ioutil.Discard, "testmusicof-bot: ", log.Lshortfile|log.LstdFlags),
+	}
+
+	err := bot.handleNominate(caller, channel)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if sentChannel != channel {
+		t.Error("Expected to send to the right channel, got", sentChannel)
```

Comment:
Missing a white space after the `got` ;)

Reply:
Not needed, formatting handles that :)